src/tree-view/keyboard-navigation/index.tsx

```diff
@@ -0,0 +1,302 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useRef } from 'react';
+import { useEffect, useMemo } from 'react';
+
+import { useStableCallback } from '@cloudscape-design/component-toolkit/internal';
+import {
+  SingleTabStopNavigationAPI,
+  SingleTabStopNavigationProvider,
+} from '@cloudscape-design/component-toolkit/internal';
+
+import AsyncStore, { useSelector } from '../../area-chart/async-store';
+import { getAllFocusables } from '../../internal/components/focus-lock/utils';
+import { KeyCode } from '../../internal/keycode';
+import handleKey, { isEventLike } from '../../internal/utils/handle-key';
+import {
+  findTreeItemByIndex,
+  findTreeItemContentById,
+  focusElement,
+  getClosestTreeItem,
+  getClosestTreeItemContent,
+  getToggleButtonOfTreeItem,
+  isElementDisabled,
+} from './utils';
+
+export function KeyboardNavigationProvider({
+  getTreeView,
+  children,
+  pageUpDownSize = 10,
```

Comment:
We used the same approach in the table so that page size can be parametrised. However, there does not seem to be the need. I recommend removing the argument in favour of a constant defined in this file.